### PR TITLE
Remove specific gun powder, and reintroduce generic one

### PR DESCRIPTION
**Pull request by Lamandus**
#### Summary
SUMMARY: Content "Remove specific gun powder, and reintroduce generic one"

#### Purpose of change
Brings back the difference between black powder and smokeless gunpowder. Since it was very inconsistend and half baked (mixed smokeless gun powder was as good as the specific ones), I went back and simplified it for gaming reasons.

### CODE REVIEW

File: data/json/obsoletion/items.json [hunk 1]
```diff
@@ -3333,5 +3333,71 @@
...
+      "gunpowder_artillery",
+      "gunpowder_large_rifle",
+      "gunpowder_rifle",
+      "gunpowder_magnum_pistol",
+      "gunpowder_pistol",
+      "gunpowder_shotgun"
+    ],
+    "type": "MIGRATION",
+    "replace": "gunpowder"
   }
```
Comment (Coolthulhu) [inline data/json/obsoletion/items.json:3402]: Should probably go to a different file. Also, I think that if a migration exists, obsolete id may be unnecessary.

Comment (Lamandus) [reply data/json/obsoletion/items.json:3402]: You mean that the definitions of the gun powders aren't needed anymore and can be deleted?

**Comment by Coolthulhu** [comment]
Mods still mention some of the gunpowders.

**Comment by Lamandus** [comment]
> Mods still mention some of the gunpowders in recipes.

done, now it has to be it.

**Pull request merged**